### PR TITLE
Update to jBCrypt 0.4 to correct an integer overflow

### PR DESCRIPTION
**Pull request by reedloden**
http://www.mindrot.org/projects/jBCrypt/news/rel04.html

### CODE REVIEW

File: ext/jruby/bcrypt_jruby/BCrypt.java [hunk 1]
```diff
@@ -747,6 +758,20 @@ public static String gensalt() {
 	 * @return	true if the passwords match, false otherwise
 	 */
 	public static boolean checkpw(String plaintext, String hashed) {
```
Comment (reedloden) [inline ext/jruby/bcrypt_jruby/BCrypt.java]: @Oscil8, does the below change end up fixing #82? Haven't had a chance to test yet, but if so, awesome...

Comment (Oscil8) [inline ext/jruby/bcrypt_jruby/BCrypt.java]: The problem with character encoding has already failed by this point -- JRuby => Java conversion of the string has already turned out of range characters into 0xfffd (Replacement character); and the use of getBytes("UTF-8") at https://github.com/reedloden/bcrypt-ruby/blob/jBCrypt-0.4/ext/jruby/bcrypt_jruby/BCrypt.java#L676 further distorts the byte values. We've implemented a workaround here https://github.com/lookout/bcrypt-ruby/tree/lookout

**Comment by reedloden** [comment]
The upstream issue is CVE-2015-0886 (also see http://osvdb.org/show/osvdb/119055).

**Comment by tjschuck** [comment]
As of https://github.com/codahale/bcrypt-ruby/pull/182, we no longer use the original jBCrypt — we now use the [spring-security fork](https://github.com/spring-projects/spring-security/blob/master/crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCrypt.java), which has since drifted away from jBCrypt 0.3 or 0.4.

**Pull request closed without merging**